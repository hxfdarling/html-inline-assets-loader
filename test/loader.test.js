
import 'jest-extended';

const { compile, evaluated } = require('./helpers');

describe('base loader', () => {
  it('should compile with `js` entry point', async () => {
    const stats = await compile('basic.js');

    expect(stats.compilation.warnings).toBeArrayOfSize(0);
    expect(stats.compilation.errors).toBeArrayOfSize(0);

    const { modules } = stats.toJson();
    const [module] = modules;
    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules))
      .toMatchSnapshot('module (evaluated)');
  });

  it('should compile with `html` entry point', async () => {
    const stats = await compile('basic.html');
    expect(stats.compilation.warnings).toBeArrayOfSize(0);
    expect(stats.compilation.errors).toBeArrayOfSize(0);

    const { modules } = stats.toJson();
    const [module] = modules;
    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules))
      .toMatchSnapshot('module (evaluated)');
  });
});
