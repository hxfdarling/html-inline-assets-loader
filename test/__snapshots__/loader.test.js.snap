// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base loader should compile with \`html\` entry point: module (evaluated) 1`] = `
"<!DOCTYPE html><html><head>
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta charset=\\"utf-8\\">
    <meta name=\\"format-detection\\" content=\\"telephone=no\\">
    <meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no\\">
    <meta itemprop=\\"image\\" content=\\"image_d56f9b.png\\">
    <link rel=\\"dns-prefetch\\" href=\\"//7.url.cn\\">
    <link rel=\\"shortcut icon \\" type=\\"image/ico\\" href=\\"image_d56f9b.png\\">
    <link rel=\\"icon\\" type=\\"image/ico\\" href=\\"image_d56f9b.png\\">
    <title>imt application</title>
  </head>

  <body>
    <script src=\\"//sqimg.qq.com/expert_qq/webrtc/WebRTCAPI_CLOUD_V3.js?v=0.0.1\\"></script>
  

</body></html>"
`;

exports[`base loader should compile with \`html\` entry point: module 1`] = `"module.exports = \\"<!DOCTYPE html><html><head>\\\\n    <meta http-equiv=\\\\\\"X-UA-Compatible\\\\\\" content=\\\\\\"IE=edge\\\\\\">\\\\n    <meta charset=\\\\\\"utf-8\\\\\\">\\\\n    <meta name=\\\\\\"format-detection\\\\\\" content=\\\\\\"telephone=no\\\\\\">\\\\n    <meta name=\\\\\\"viewport\\\\\\" content=\\\\\\"width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no\\\\\\">\\\\n    <meta itemprop=\\\\\\"image\\\\\\" content=\\\\\\"image_d56f9b.png\\\\\\">\\\\n    <link rel=\\\\\\"dns-prefetch\\\\\\" href=\\\\\\"//7.url.cn\\\\\\">\\\\n    <link rel=\\\\\\"shortcut icon \\\\\\" type=\\\\\\"image/ico\\\\\\" href=\\\\\\"image_d56f9b.png\\\\\\">\\\\n    <link rel=\\\\\\"icon\\\\\\" type=\\\\\\"image/ico\\\\\\" href=\\\\\\"image_d56f9b.png\\\\\\">\\\\n    <title>imt application</title>\\\\n  </head>\\\\n\\\\n  <body>\\\\n    <script src=\\\\\\"//sqimg.qq.com/expert_qq/webrtc/WebRTCAPI_CLOUD_V3.js?v=0.0.1\\\\\\"></script>\\\\n  \\\\n\\\\n</body></html>\\";"`;

exports[`base loader should compile with \`js\` entry point: module (evaluated) 1`] = `
"<!DOCTYPE html><html><head>
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta charset=\\"utf-8\\">
    <meta name=\\"format-detection\\" content=\\"telephone=no\\">
    <meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no\\">
    <title>imt application</title>
  </head>

  <body>

</body></html>"
`;

exports[`base loader should compile with \`js\` entry point: module 1`] = `"module.exports = \\"<!DOCTYPE html><html><head>\\\\n    <meta http-equiv=\\\\\\"X-UA-Compatible\\\\\\" content=\\\\\\"IE=edge\\\\\\">\\\\n    <meta charset=\\\\\\"utf-8\\\\\\">\\\\n    <meta name=\\\\\\"format-detection\\\\\\" content=\\\\\\"telephone=no\\\\\\">\\\\n    <meta name=\\\\\\"viewport\\\\\\" content=\\\\\\"width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no\\\\\\">\\\\n    <title>imt application</title>\\\\n  </head>\\\\n\\\\n  <body>\\\\n\\\\n</body></html>\\";"`;
